### PR TITLE
update puppeteer for headless

### PR DESCRIPTION
**Pull request by jywarren**
Re https://github.com/karma-runner/karma-chrome-launcher/issues/158#issuecomment-396914961 and https://github.com/publiclab/image-sequencer-app/issues/6, trying this out!

### CODE REVIEW

**Comment by VibhorCodecianGupta** [comment]
@jywarren also try adding the arg `'--disable-setuid-sandbox`. Not certain that it'd work, but it did for some people over at [this issue](https://github.com/GoogleChrome/puppeteer/issues/807)
Related discussion at #1032

**Comment by jywarren** [comment]
Oh this would be great to try. Thanks @VibhorCodecianGupta. I'm a bit occupied this week with our staff retreat meetings, but if someone else wants to try this please go for it!

File: src/modules/_nomodule/gl-context.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = function runInBrowserContext(input, callback, step, options) {
 
     var obj = { input: input, modOptions: minOptions }
 
-    puppeteer.launch().then(function(browser) {
+    puppeteer.launch({headless: true, args:['--no-sandbox --disable-setuid-sandbox']}).then(function(browser) {
```
Comment (jywarren) [inline src/modules/_nomodule/gl-context.js]: Tried a couple new things here. Maybe we should just merge this if it passes and then see if it works upstream? @VibhorCodecianGupta @HarshKhandeparkar @Divy123 ?

Comment (VibhorCodecianGupta) [reply src/modules/_nomodule/gl-context.js]: Sounds good. I'd test this but I do not have a GCF instance. `image-sequencer-app` is deployed on GCF, right @jywarren ?

Comment (Divy123) [reply src/modules/_nomodule/gl-context.js]: @jywarren I also do not have GCF instance. Any suggestions on testing it?

Comment (HarshKhandeparkar) [reply src/modules/_nomodule/gl-context.js]: We can test it on some cloud service like Google Cloud or firebase just to make sure it isn't broken

Comment (jywarren) [reply src/modules/_nomodule/gl-context.js]: Thanks, i just added @Divy123 and @VibhorCodecianGupta -- we're using this in Cloud Functions:

https://gist.github.com/jywarren/7721a38725affdbc4fad2598ff1be59e

@HarshKhandeparkar i can add you too if you want to send me your email address. 

Thanks!

Comment (jywarren) [reply src/modules/_nomodule/gl-context.js]: More info here: https://github.com/publiclab/image-sequencer-app/issues/6

Thanks, all!

Comment (VibhorCodecianGupta) [reply src/modules/_nomodule/gl-context.js]: @jywarren amazing, I'll try running it.
 
Also, I think issues #1030 #1033 are pretty urgent, I listed some possible fixes and your review is required there. Plugging in my proposal updates to your response as well (hope that's fine! :D) https://bit.ly/2VlDshL

**Comment by jywarren** [comment]
Using: 

* https://rominirani.com/using-puppeteer-in-google-cloud-functions-809a14856e14
* https://cloud.google.com/blog/products/gcp/introducing-headless-chrome-support-in-cloud-functions-and-app-engine

**Comment by HarshKhandeparkar** [comment]
@jywarren please review #1038. I will have to rebase now?

**Pull request merged**